src/util/message.ts

```diff
@@ -0,0 +1,95 @@
+import { decodeJWT } from '@tonomy/did-jwt';
+// import { issue, OutputType } from '@tonomy/antelope-ssi-toolkit';
+import { Issuer, verifyCredential, W3CCredential } from '@tonomy/did-jwt-vc';
+
+// import { Resolver } from '@tonomy/did-resolver';
+import { getSettings } from '../settings';
+import { JWTDecoded } from '@tonomy/did-jwt/lib/JWT';
+import crossFetch from 'cross-fetch';
+import { getResolver } from '@tonomy/antelope-did-resolver';
+import { Resolver } from '@tonomy/did-resolver';
+import { issue, OutputType } from '@tonomy/antelope-ssi-toolkit';
+import { resolve } from './did-jwk';
+
+export class Message {
+    private decodedJwt: JWTDecoded;
+
+    /**
+     * creates a signed message and return message object
+     * @param message the messageResolver with the signer and the did
+     * @param recipient the recipient id
+     * @returns a message objects
+     */
+    static async sign(message: object, issuer: Issuer, recipient?: string): Promise<Message> {
+        const vc: W3CCredential = {
+            '@context': ['https://www.w3.org/2018/credentials/v1'],
+            // should this be tonomy foundation ?
+            id: 'https://example.com/id/1234324',
+            type: ['VerifiableCredential'],
+            issuer: {
+                id: issuer.did,
+            },
+            issuanceDate: new Date().toISOString(),
+            credentialSubject: {
+                message,
```

Comment:
add a `type: string` property here that lets subscribers know how to treat messages
e.g. `type: "tonomyIdLogin"` or "websiteLoginRequest"
New Task

Reply:
this type can be used in the `subscribe()` functions as a filter, so that different components can subscribe to messages of a certain type